flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManager.java

```diff
@@ -185,6 +199,10 @@ public ResourceManager(
 		this.jmResourceIdRegistrations = new HashMap<>(4);
 		this.taskExecutors = new HashMap<>(8);
 		this.taskExecutorGatewayFutures = new HashMap<>(8);
+
+		this.defaultTaskManagerMemoryMB = ConfigurationUtils.getTaskManagerHeapMemory(flinkConfig).getMebiBytes();
+		this.numberOfTaskSlots = flinkConfig.getInteger(TaskManagerOptions.NUM_TASK_SLOTS);
+		this.slotsPerWorker = updateTaskManagerConfigAndCreateWorkerSlotProfiles(this.flinkConfig, defaultTaskManagerMemoryMB, numberOfTaskSlots);
```

Comment:
Can we move these computations out of the `ResourceManager`? It seems as if we only rely on static helper methods. We could create a `TaskManagerSpecification` object containing the required information. That way, it would also be easier to test the correct computation without having to introduce these `VisibleForTesting` methods and having to create `RpcEndpoints` for only testing static helper methods.